src/Executables/ReduceCceWorldtube/ReduceCceWorldtube.cpp

```diff
@@ -0,0 +1,296 @@
+void perform_cce_worldtube_reduction(const std::string& input_file,
+                                     const std::string& output_file,
+                                     const size_t buffer_depth) noexcept {
+  Cce::SpecWorldtubeH5BufferUpdater buffer_updater{input_file};
+  const size_t l_max = buffer_updater.get_l_max();
+  // Perform the boundary computation to scalars at twice the input l_max to be
+  // absolutely certain that there are no problems associated with aliasing.
+  const size_t computation_l_max = 2 * l_max;
+
+  // we're not interpolating, this is just a reasonable number of rows to ingest
+  // at a time.
+  const size_t size_of_buffer = square(l_max + 1) * (buffer_depth);
+  const DataVector& time_buffer = buffer_updater.get_time_buffer();
+
+  Variables<Cce::detail::cce_input_tags> coefficients_buffers{size_of_buffer};
+  Variables<Cce::detail::cce_input_tags> coefficients_set{
+      Spectral::Swsh::size_of_libsharp_coefficient_vector(computation_l_max)};
+
+  using boundary_variables_tag =
+      Tags::Variables<Cce::Tags::characteristic_worldtube_boundary_tags>;
+
+  auto boundary_data_box =
+      db::create<db::AddSimpleTags<boundary_variables_tag>>(
+          db::item_type<boundary_variables_tag>{
+              Spectral::Swsh::number_of_swsh_collocation_points(
+                  computation_l_max)});
+
+  using reduced_boundary_tags =
+      tmpl::list<Cce::Tags::BoundaryValue<Cce::Tags::BondiBeta>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiU>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiQ>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiW>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiJ>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::Dr<Cce::Tags::BondiJ>>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiH>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::BondiR>,
+                 Cce::Tags::BoundaryValue<Cce::Tags::DuRDividedByR>>;
+
+  size_t time_span_start = 0;
+  size_t time_span_end = 0;
+  Cce::ReducedWorldtubeModeRecorder recorder{output_file};
+
+  ComplexModalVector output_goldberg_mode_buffer{square(computation_l_max + 1)};
+  ComplexModalVector output_libsharp_mode_buffer{
+      Spectral::Swsh::size_of_libsharp_coefficient_vector(computation_l_max)};
+
+  for (size_t i = 0; i < time_buffer.size(); ++i) {
+    const double time = time_buffer[i];
+    std::cout << "reduceing data at time : " << time << " / "
+              << time_buffer[time_buffer.size() - 1] << "\r" << std::flush;
+    buffer_updater.update_buffers_for_time(
+        make_not_null(&coefficients_buffers), make_not_null(&time_span_start),
+        make_not_null(&time_span_end), time, 0, buffer_depth);
+
+    slice_buffers_to_libsharp_modes(
+        make_not_null(&coefficients_set), coefficients_buffers,
+        time_span_end - time_span_start, i - time_span_start, l_max,
+        computation_l_max);
+
+    Cce::create_bondi_boundary_data(
```

Comment:
where is this function defined; there are too many functions with this name 😢

Reply:
This one is in `BoundaryData.hpp`, but the question makes me realize I've mistakenly omitted the if for dispatching to the unnormalized version.